server/chat-plugins/random-battles.ts

```diff
@@ -244,6 +244,377 @@ function CAP1v1Sets(species: string | Species) {
+					evos.push(`${evo.name} (level-up with high Friendship${condition})`);
+					break;
+				case 'levelHold':
+					evos.push(`${evo.name} (level-up holding ${evo.evoItem}${condition})`);
+					break;
+				case 'useItem':
+					evos.push(`${evo.name} (${evo.evoItem})`);
+					break;
+				case 'levelMove':
+					evos.push(`${evo.name} (level-up with ${evo.evoMove}${condition})`);
+					break;
+				case 'other':
+					evos.push(`${evo.name} (${evo.evoCondition})`);
+					break;
+				case 'trade':
+					evos.push(`${evo.name} (trade${evo.evoItem ? ` holding ${evo.evoItem}` : condition})`);
+					break;
+				default:
+					evos.push(`${evo.name} (${evo.evoLevel}${condition})`);
+				}
+			}
+		}
+		if (!evos.length) {
+			details[`<font color="#686868">Does Not Evolve</font>`] = "";
+		} else {
+			details["Evolution"] = evos.join(", ");
+		}
+		buf += `<font size="1">${Object.keys(details).map(detail => {
+			if (details[detail] === '') return detail;
+			return `<font color="#686868">${detail}:</font> ${details[detail]}`;
+		}).join("&nbsp;|&ThickSpace;")}</font>`;
+	}
+	return buf;
+}
+
+function generateSSBInnateInfo(name: string, dex: ModdedDex, baseDex: ModdedDex) {
+	let buf = ``;
+	// Special casing for users whose usernames are already existing, i.e. Perish Song
+	let effect = dex.getEffect(name + 'user');
+	let longDesc = ``;
+	const baseAbility = Dex.deepClone(baseDex.getAbility('noability'));
+	// @ts-ignore hack to record the name of the innate abilities without using name
+	if (effect.exists && effect.innateName && (effect.desc || effect.shortDesc)) {
+		// @ts-ignore hack
+		baseAbility.name = effect.innateName;
+		if (effect.desc) baseAbility.desc = effect.desc;
+		if (effect.shortDesc) baseAbility.shortDesc = effect.shortDesc;
+		buf += `<hr />Innate Ability:<br />${Chat.getDataAbilityHTML(baseAbility)}`;
+		if (effect.desc && effect.shortDesc && effect.desc !== effect.shortDesc) {
+			longDesc = effect.desc;
+		}
+	} else {
+		effect = dex.getEffect(name);
+		// @ts-ignore hack
+		if (effect.exists && effect.innateName && (effect.desc || effect.shortDesc)) {
+			// @ts-ignore hack
+			baseAbility.name = effect.innateName;
+			if (effect.desc) baseAbility.desc = effect.desc;
+			if (effect.shortDesc) baseAbility.shortDesc = effect.shortDesc;
+			buf += `<hr />Innate Ability:<br />${Chat.getDataAbilityHTML(baseAbility)}`;
+			if (effect.desc && effect.shortDesc && effect.desc !== effect.shortDesc) {
+				longDesc = effect.desc;
+			}
+		}
+	}
+	if (buf) {
+		const details: {[k: string]: string} = {
+			Gen: '8',
+		};
+		buf += `<font size="1">${Object.keys(details).map(detail => {
+			if (details[detail] === '') return detail;
+			return `<font color="#686868">${detail}:</font> ${details[detail]}`;
+		}).join("&nbsp;|&ThickSpace;")}</font>`;
+	}
+	if (longDesc) {
+		buf += `<details><summary><strong>In-Depth Description</strong></summary>${longDesc}</details>`;
+	}
+	return buf;
+}
+
+function SSBSets(target: string) {
```

Comment:
something something camelCase this + ssbSets is too vague for this function, maybe "getSsbSet"?

Reply:
@KrisXV

Reply:
Not going to camelCase this, and if you view at the other functions in the file like this they're all under the same naming scheme.

Reply:
Well, to clarify since they're not under the same naming scheme; there is no naming scheme. However, there are other functions under this naming scheme and this is a discussion for another PR that im making soon.